plugins/DevicesDetection/Columns/BrowserName.php

```diff
@@ -21,7 +21,7 @@
 class BrowserName extends Base
 {
     protected $columnName = 'config_browser_name';
-    protected $columnType = 'VARCHAR(10) NULL';
+    protected $columnType = 'VARCHAR(40) NULL';
```

Comment:
Just to better understand can you mention what values be like? Was thinking ideally this was referencing `log_action` so we don't always store the full name but that would mean migrating all existing values which isn't really doable / realistic. I guess we don't want to use any mappings having of `short_name` to `long_name` considering we would have 10 characters available that would give us quite a bit of flexibility already?

Reply:
Device detector is able to detect various client names, and there are also some dynamic detection for e.g. mobile apps. You can see a full list here: https://github.com/matomo-org/device-detector/blob/f33d196a7790689f721db64766f5ec3eec470284/README.md#list-of-detected-libraries

I would love to go away from the short codes at all at some point, and having it in `log_action` would be awesome. But as you mentioned a migration isn't easily doable.
We could alternatively also add another id column and start storing the client names in log_action separately. We could than maybe switch on using those data for creating the reports in a later release when we already have some data logged, so we maybe don't need too much logic to merge it with the old reports. What do you think?

Reply:
@sgiehl I'd say really depends how easy it is to do... I suppose a segment would then check using `OR` either in the column or in the log_action? Let's maybe for now just keep using the VARCHAR then

Reply:
@tsteur adding and filling that new column is very easy and quickly done. That way we wouldn't also change any of the existing reports & segments. New reports and changes to segments could then be done any time later

Reply:
Up to you @sgiehl I reckon either is fine

Reply:
Will change it then. Makes more sense to have a new id column. Will check how easy we can change it so only the new column is used for fresh Matomo installs...

Reply:
@tsteur just had another look at that, and realized, that the client actually is no action dimension or shouldn't be one. So all the action id magic wouldn't work out of the box. Also we would need to reserve a action type for it, even though it isn't an action at all. So it might be a bit more effort than I thought... But guess it would still be the better solution to not store the full names in visit table 🤔

Reply:
Let's just store it in full text for now. Seems not worth the effort. If the tests pass feel free to merge @sgiehl